### PR TITLE
Fix most npm audit errors, exclude advisory 1556

### PR DESCRIPTION
**Pull request by diracdeltas**
Resolves 28 out of 33 npm audit errors by rerunning npm install.
https://www.npmjs.com/advisories/1556 is ignored because there is no fix
in some of our dependencies, and it's a low-impact DoS vulnerability.

Fix https://github.com/brave/brave-browser/issues/11732

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

* `npm run audit_deps` should exit cleanly
* `npm run audit_deps -- --audit_dev_deps` should also exit cleanly

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: script/audit_deps.py [hunk 1]
```diff
@@ -88,26 +84,30 @@ def npm_audit_deps(path, args):
...
 
     resolutions, non_dev_exceptions = extract_resolutions(result)
 
-    # Trigger a failure if there are non-dev exceptions
-    if non_dev_exceptions:
+    if resolutions:
         print('Result: Audit finished, vulnerabilities found')
+
+    # Trigger a failure if there are non-dev exceptions
+    if non_dev_exceptions and not args.audit_dev_deps:
```
Comment (fmarier) [inline script/audit_deps.py:100]: In the case that there are `non_dev_exceptions` and that `args.audit_dev_deps` is set, we seem to be ignoring the non-dev exceptions. Is that intended?

If I'm reading this right then if `--audit_dev_deps` is passed and there are only non-dev exceptions, we end up returning success (`0`).

File: script/audit_deps.py [hunk 2]
```diff
@@ -21,7 +21,9 @@
     os.path.join('vendor', 'brave-extension', 'node_modules'),
 ]
 
+# Ping security team before adding to ignored_npm_advisories
```
Comment (fmarier) [inline script/audit_deps.py:24]: :100:

File: script/audit_deps.py (hunk 1 above)
Comment (diracdeltas) [reply script/audit_deps.py:100]: @fmarier i think you may have missed a `not` in the if condition here? if `args.audit_dev_deps` is true, and there are `non_dev_exceptions`, `resolutions` is also non-empty (because it's a superset of non_dev_exceptions) so we should be returning 1 here: https://github.com/brave/brave-core/pull/6661/files#diff-1539a0feacaeda0bff2a27b87da0a165R107

but now that i think about it, this block (where non_dev_exceptions exist and `args.audit_dev_deps` is false) should return 0, not 1. 

to summarize the intended behavior:
* if there are any exceptions and audit_dev_deps is passed, return 1
* if there are dev exceptions only and audit_dev_deps is NOT passed, return 0
* if there are any non-dev exceptions and audit_dev_deps is NOT passed, return 1
* if there are no exceptions, return 0

Comment (fmarier) [reply script/audit_deps.py:100]: > resolutions is also non-empty (because it's a superset of non_dev_exceptions)

Ah, that's the part I missed, I hadn't realized that `resolutions` contained both.

**Review by fmarier — APPROVED** [review]

File: script/audit_deps.py (hunk 1 above)
Comment (diracdeltas) [reply script/audit_deps.py:100]: i think so - https://github.com/brave/brave-core/pull/6661/files#diff-1539a0feacaeda0bff2a27b87da0a165R154

**Comment by mihaiplesa** [comment]
Is the Travis fail expected?

**Comment by diracdeltas** [comment]
nope, looking into the travis failure right now

**Comment by diracdeltas** [comment]
Travis is passing now, and MacOS CI succeeded. The Win/Linux builds are failing with `An error occurred (InsufficientInstanceCapacity) when calling the StartInstances operation (reached max retries: 4): Insufficient capacity.` in the start-node step. Seems unrelated?

**Comment by diracdeltas** [comment]
restarted it and CI on windows passed
